kyaml/kio/byteio_writer.go

```diff
@@ -90,9 +94,8 @@ func (w ByteWriter) Write(nodes []*yaml.RNode) error {
 	// don't wrap the elements
 	if w.WrappingKind == "" {
 		for i := range nodes {
-			err := encoder.Encode(nodes[i].Document())
-			if err != nil {
-				return errors.Wrap(err)
+			if err := w.encode(encoder, nodes[i]); err != nil {
```

Comment:
Would you write a unit test for this to make sure a collection of Nodes gets properly serialized as json.

e.g.

```json
{
  {...},
  {...}
}
```